sdks/python/apache_beam/io/localfilesystem.py

```diff
@@ -0,0 +1,206 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+"""Local File system for file based sources and sinks."""
+
+from __future__ import absolute_import
+
+import glob
+import os
+import shutil
+
+from apache_beam.io.filesystem import CompressedFile
+from apache_beam.io.filesystem import CompressionTypes
+from apache_beam.io.filesystem import FileSystem
+from apache_beam.io.filesystem import FileMetadata
+
+
+class LocalFileSystem(FileSystem):
+  """A local file system implementation for accessing files on disk.
+  """
+
+  def mkdirs(self, path):
+    """Recursively create directories for the provided path.
+
+    Args:
```

Comment:
Ditto about docs (I don't think all-subclasses should copy doc comments from the super class).